gcloud-java-dns/src/main/java/com/google/gcloud/dns/BatchResultImpl.java

```diff
@@ -0,0 +1,44 @@
+package com.google.gcloud.dns;
+
+import com.google.gcloud.dns.batch.BatchResult;
+
+/**
+ * This class holds a single result of a batch call to the Cloud DNS.
+ */
+public class BatchResultImpl<T> implements BatchResult<T, DnsException> {
```

Comment:
As you said, this should we should have the interface or base class in core, however it may make
sense to subclass it in DNS (to avoid the need to specify the generic type by the user).